### PR TITLE
Add RequiredConversationResolution protection field

### LINKED ISSUE #1945: Missed required_conversation_resolution field in branch protection rules
GitHub added a new option for branch protection rules: [Require conversation resolution before merging](https://docs.github.com/en/github/administering-a-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-conversation-resolution-before-merging), v3 API also includes this new field as `required_conversation_resolution` (see example below). But current version (v36) of go-github doesn't include this field in [protection](https://github.com/google/go-github/blob/master/github/repos.go#L753) struct. 

Example API call:
```json
{
  "url": "https://api.github.com/repos/g4s8/gitstrap/branches/master/protection", 
  "required_status_checks": {
    "url": "https://api.github.com/repos/g4s8/gitstrap/branches/master/protection/required_status_checks",
    "strict": true,
    "contexts": [
      "build",
      "lint",
      "test"
    ],
    "contexts_url": "https://api.github.com/repos/g4s8/gitstrap/branches/master/protection/required_status_checks/contexts"
  },
  "required_pull_request_reviews": {
    "url": "https://api.github.com/repos/g4s8/gitstrap/branches/master/protection/required_pull_request_reviews",
    "dismiss_stale_reviews": true,
    "require_code_owner_reviews": false,
    "required_approving_review_count": 1
  },
  "enforce_admins": {
    "url": "https://api.github.com/repos/g4s8/gitstrap/branches/master/protection/enforce_admins",
    "enabled": false
  },
  "required_linear_history": {
    "enabled": true
  },
  "allow_force_pushes": {
    "enabled": false
  },
  "allow_deletions": {
    "enabled": false
  },
  "required_conversation_resolution": {
    "enabled": false
  }
}
```

### PR DESCRIPTION
**Pull request by g4s8**
Fixes #1945 - added a new field for `Protection` struct
to parse `required_conversation_resolution` JSON field
from API response for branch protection rules.
Updated test.

### CODE REVIEW

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:853]:
```suggestion
// RequiredConversationResolution, if enabled, requires all comments on the pull request to be resolved before it can be merged to a protected branch.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @g4s8 !
Just a couple minor tweaks, please, for documentation purposes. (Note the ending period.)

**Comment by g4s8** [comment]
@gmlewis thanks, fixed in https://github.com/google/go-github/pull/1946/commits/6a691240f1db8e6131b8fef437c8da07e9e8a89d

**Review by gmlewis — APPROVED** [review]
Thank you, @g4s8 !
LGTM.

Awaiting second LGTM before merging.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Pull request merged**
